ssl4rs/models/classif/base.py

```diff
@@ -367,3 +368,138 @@ def _render_and_log_samples(
+        save_hyperparams: bool = True,  # turn this off in derived classes
+        **kwargs,
+    ):
+        if save_hyperparams:
+            # this line allows us to access hparams with `self.hparams` + auto-stores them in checkpoints
+            self.save_hyperparameters(logger=False)  # logger=False since we don't need duplicated logs
+
+        super().__init__(
+            model=model,
+            loss_fn=loss_fn,
+            metrics=None,
+            # inits the segmenter base model with None but also overiide the configure metrics method below to allow for masked MSE
+            optimization=optimization,
+            num_output_classes=1,  # only one channel as it is a regression objective
+            num_input_channels=num_input_channels,
+            input_key=input_key,
+            label_key=label_key,
+            ignore_index=ignore_index,
+            example_image_shape=example_image_shape,  # height, width
+            save_hyperparams=save_hyperparams,  # turn this off in derived classes
+            **kwargs,
+        )
+
+        loss_fn = MaskedMSELoss(ignore_index=ignore_index)
+        assert isinstance(loss_fn, torch.nn.Module), f"incompatible loss_fn type: {type(loss_fn)}"
+        self.loss_fn = loss_fn
+        self.num_output_classes = 1
+
+    def configure_metrics(self):
+        return torchmetrics.MetricCollection({'masked_mse': MaskedMeanSquaredError(ignore_index=-1)})
+
+    def _render_and_log_samples(
+        self,
+        loop_type: str,  # 'train', 'valid', or 'test'
+        batch: ssl4rs.data.BatchDictType,
+        batch_idx: int,
+        sample_idxs: typing.List[int],
+        sample_ids: typing.List[typing.Hashable],
+        outputs: typing.Dict[typing.AnyStr, typing.Any],
+        dataloader_idx: int = 0,
+        fill_value: float = 0.9,
+        ignore_idx: int = -1
+    ) -> typing.Any:
+        """Renders and logs specific samples from the current batch using available loggers.
+
+        Note: only available when using binary classification models/masks.
+        """
+        if self.num_output_classes != 1:
+            return None  # not clear how to render here, let's let derived classes handle it
+        assert len(sample_idxs) == len(sample_ids) and len(sample_idxs) > 0
+        batch_size = ssl4rs.data.get_batch_size(batch)
+        preds, targets = outputs["preds"], outputs["targets"]
+        assert targets.ndim == 4 and targets.shape[0] == batch_size and targets.dtype == torch.float32
+        tensor_shape = targets.shape[2:]
+        assert preds.ndim == 4 and preds.shape == (batch_size, self.num_output_classes, *tensor_shape)
+
+        # we'll render the input tensors with a prediction mask and target mask side-by-side
+        outputs = []
+        for sample_idx, sample_id in zip(sample_idxs, sample_ids):
+            input_tensor = batch[self.input_key][sample_idx].cpu()
+            img_rgb = batch['location_preview_image'][sample_idx].cpu()
```

Comment:
this makes the class DISA-specific, so many it should be in a disa subfolder or something